### PR TITLE
implement the percentile point functions (inverse of cdf)

### PR DESCRIPTION
**Pull request by segasai**
Hi, 

I've implemented the PPF (percentile point function; inverse of CDF) for all the distributions. They are useful for example for sampling (i.e. nested sampling where you need to convert your prior into uniform from zero to one). 
I've also added tests that they work as inverse as CDFs. 

   Sergey

### CODE REVIEW

File: imf/distributions.py [hunk 1]
```diff
@@ -156,6 +172,22 @@ def rvs(self, N):
...
+        assert(x1.max()<=1)
+        edges = np.r_[[0],np.cumsum(self.weights)]
+        # edges of powerlaw in CDF scale from 0 to 1
+        pos = np.digitize(x1, edges) # bin positions, 1 is the leftmost 
+        left = edges[pos-1]
+        w = self.weights[pos-1]
+        x2 = np.clip((x1-left)/w,0,1) # mapping to 0,1 on the segment
+        ret =np.zeros_like(x1)
+        for ii in range(x.size):
+            ret[ii] = self.pows[pos[ii]-1].ppf(x2[ii])
```
Comment (keflavich) [inline imf/distributions.py]: this might be more efficiently implemented as a list comprehension

Comment (segasai) [reply imf/distributions.py]: I agree that the line is not the most efficient, but I don't think there is a way around that -- multiple points may correspond to different segments. Regarding the list comprehension, to my knowledge that'll be just stylistic difference. I'm not expecting any performance difference.
I can imagine a faster version (if input is long) is to group by segment. But I thought I first implement a trivial version and if it is ever a problem on can optimize it later.

Comment (keflavich) [reply imf/distributions.py]: the list comprehension is a little more than a stylistic difference, but it's unclear whether there would be a performance benefit in this case, so I agree, leave it as is.

**Comment by segasai** [comment]
Let me also push one more commit before merging, which ensures that nan's are returned when input is outside 0,1 range

**Comment by segasai** [comment]
I think I've concluded my push request now

**Pull request merged**